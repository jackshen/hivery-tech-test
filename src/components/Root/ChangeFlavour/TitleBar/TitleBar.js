import React from "react";
import styled from "styled-components";

const TitleBarWrapper = styled.div`
  border-bottom: 0.125rem solid ${props => props.theme.solitude};
  box-sizing: border-box;
  font-size: 1.25rem;
  grid-area: title-bar;
  padding: 1.5rem 2.4375rem;
  width: 100%;
`;

const TitleBar = () => {
  return <TitleBarWrapper>Change Flavour</TitleBarWrapper>;
};

export default TitleBar;
