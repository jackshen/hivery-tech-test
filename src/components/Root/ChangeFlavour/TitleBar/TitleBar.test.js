// including purely a snapshot test like this has its advantages and disadvantages
// the main advantage being that accidental changes to things such as the Styled Components' theme
// would not cause stylistic breaks

import React from "react";

import wrapAll from "#test/wrapAll";

import TitleBar from "./TitleBar";

describe("TitleBar", () => {
  test("matches snapshot", () => {
    const { container } = wrapAll(<TitleBar />);

    expect(container).toMatchSnapshot();
  });
});
