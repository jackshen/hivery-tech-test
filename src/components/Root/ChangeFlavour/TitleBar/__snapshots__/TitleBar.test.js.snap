// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar matches snapshot 1`] = `
.c0 {
  border-bottom: 0.125rem solid #e4e6ea;
  box-sizing: border-box;
  font-size: 1.25rem;
  grid-area: title-bar;
  padding: 1.5rem 2.4375rem;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    Change Flavour
  </div>
</div>
`;
